pages/index.tsx

```diff
@@ -44,7 +44,9 @@ export async function getStaticProps({
   // We prefer to do the computation at buildtime/servertime
   const { featured, bestSelling } = (() => {
     // Create a copy of products that we can mutate
+    // Filter products that do not have images
     const products = [...newestProducts]
+      .filter((p) => p.node.images.edges!.length > 0)
```

Comment:
I'd rather not add another iteration over the products array. What about rendering conditionally the image component or a placeholder. (The more computations we can avoid, the better)